pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -132,19 +132,18 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			}(),
 		})
 		deployInternalDomainDNSRecord = g.Add(flow.Task{
-			Name:         "Deploying internal domain DNS record",
-			Fn:           flow.TaskFn(botanist.DeployInternalDomainDNSRecord).DoIf(dnsEnabled && managedInternalDNS && !o.Shoot.HibernationEnabled),
+			Name: "Deploying internal domain DNS record",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
```

Comment:
```suggestion
			Fn:           flow.TaskFn(botanist.InternalDNS().Deploy).DoIf(!o.Shoot.HibernationEnabled),
```

Reply:
Some of those DNSEntries can change their values therefore (by default they are not set ad only set once the APIServer address is correct.

Reply:
Can you workaround it with
```suggestion
			Fn:           flow.TaskFn(botanist.DeployInternalDNS).DoIf(!o.Shoot.HibernationEnabled),
```

and
```golang
func (b *Botanist) DeployInternalDNS(ctx context.Context) error {
	return b.InternalDNS().Deploy(ctx)
}
```
?